### PR TITLE
Render operations by type, improve render blocks

### PR DESCRIPTION
**Pull request by aristidesstaffieri**
Motivation:
Previously we displayed operations by digging into their private fields and conditionally rendering all fields that could be on any op type. This caused us to drop the operation type and made it difficult to know how any 1 operation would be rendered. It also made it more difficult to get into the auth tree and sub-invocations.

Changes:
Splits rendering by operation type.
Extracts all KeyVals into a new file, adds new KeyVals for missing types.
Adds missing fields, and fixes rendering bugs for a few types.

### CODE REVIEW

File: extension/src/popup/components/signTransaction/Operations/KeyVal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+  function renderSignerType() {
+    if ("ed25519PublicKey" in signer) {
+      return (
+        <KeyValueList
+          operationKey={t("Signer")}
+          operationValue={signer.ed25519PublicKey}
+        />
+      );
+    }
+    if ("sha256Hash" in signer) {
```
Comment (quietbits) [inline extension/src/popup/components/signTransaction/Operations/KeyVal/index.tsx:163]: Nit: space above this `if` statement. The other ones have it.

File: extension/src/popup/components/signTransaction/Operations/KeyVal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+
+  if ("ed25519PublicKey" in signer) {
+    return (
+      <KeyValueWithPublicKey
+        operationKey={t("Signer Key")}
+        operationValue={signer.ed25519PublicKey}
+      />
+    );
+  }
+  if ("sha256Hash" in signer) {
```
Comment (quietbits) [inline extension/src/popup/components/signTransaction/Operations/KeyVal/index.tsx:266]: Nite: same as above about the space. :policewoman:

File: extension/src/popup/views/SignTransaction/index.tsx [hunk 3]
```diff
@@ -78,6 +78,7 @@ export const SignTransaction = () => {
     isHttpsDomain,
     flaggedKeys,
   } = tx;
+  console.log(Trans, isHttpsDomain);
```
Comment (quietbits) [inline extension/src/popup/views/SignTransaction/index.tsx:81]: Do we need this `console.log` here?

File: extension/src/popup/views/SignTransaction/index.tsx [hunk 4]
```diff
@@ -200,24 +201,24 @@ export const SignTransaction = () => {
...
+  //       <p>
+  //         <Trans domain={domain}>
+  //           The website <strong>{{ domain }}</strong> does not use an SSL
+  //           certificate. For additional safety Freighter only works with
+  //           websites that provide an SSL certificate.
+  //         </Trans>
+  //       </p>
+  //     </WarningMessage>
+  //   );
+  // }
```
Comment (quietbits) [inline extension/src/popup/views/SignTransaction/index.tsx:221]: Should we remove this?

**Review by quietbits — APPROVED** [review]

File: extension/src/popup/views/SignTransaction/index.tsx (hunk 4 above)
Comment (aristidesstaffieri) [reply extension/src/popup/views/SignTransaction/index.tsx]: whoops! This was commented out to test an API path locally, commented back in https://stellarfoundation.slack.com/archives/C06GA82KU2H/p1707081896826859?thread_ts=1706810112.655399&cid=C06GA82KU2H

**Pull request merged**